tests/index_test.js

```diff
@@ -9,12 +9,205 @@
+  validatorInstance.validate(['appveyur.yml']);
+
+  test.equal(validatorInstance.logs.length, 1);
+  test.equal(validatorInstance.inValidFilesCount, 1);
+});
+
+tape('Wrong filepath #2', test => {
+  test.plan(2);
+
+  const validatorInstance = new validator();
+  validatorInstance.validate(['appveyur.yml', './appveyur.yml', '/var/www/sample.yml']);
+
+  test.equal(validatorInstance.logs.length, 3);
+  test.equal(validatorInstance.inValidFilesCount, 3);
+});
+
+tape('report() increments log length by one if no invalid files', test => {
+  test.plan(2);
+
+  const validatorInstance = new validator();
+  validatorInstance.validate(['appveyor.yml']);
+
+  test.equal(validatorInstance.logs.length, 0);
+
+  validatorInstance.report();
+
+  test.equal(validatorInstance.logs.length, 1);
+});
+
+tape('report() increments log length by two if invalid files present', test => {
+  test.plan(2);
+
+  const validatorInstance = new validator();
+  validatorInstance.validate(['appveyur.yml']);
+
+  test.equal(validatorInstance.logs.length, 1);
+
+  validatorInstance.report();
+
+  test.equal(validatorInstance.logs.length, 3);
+});
+
+tape('Valid Structure', test => {
+  test.plan(1);
+
+  const options = {
+    structure: {
+      environment: {
+        matrix: [
+          {
+            nodejs_version: 'string'
+          }
+        ]
+      },
+      version: 'string',
+      init: [
+        'string'
+      ],
+      clone_depth: 'number',
+      matrix: {
+        fast_finish: 'boolean'
+      },
+      cache: [
+        'string'
+      ],
+      install: [
+        {
+          ps: 'string'
+        },
+        'string'
+      ],
+      test_script: [
+        'string'
+      ],
+      build: 'string'
+    }
+  }
+
+  const validatorInstance = new validator(options);
+  validatorInstance.validate(['appveyor.yml']);
+
+  test.equal(validatorInstance.inValidFilesCount, 0);
+});
+
+tape('Invalid Structure', test => {
+  test.plan(1);
+
+  const options = {
+    structure: {
+      environment: {
+        matrix: [
+          {
+            nodejs_version: 'string'
+          }
+        ]
+      },
+      init: [
+        'string'
+      ],
+      version: 'string',
+      clone_depth: 'number',
+      matrix: {
+        fast_finish: 'number'
+      },
+      cache: [
+        'string'
+      ],
+      install: [
+        {
+          ps: 'string'
+        },
+        'string'
+      ],
+      test_script: [
+        'string'
+      ],
+      build: 'string'
+    }
+  }
+
+  const validatorInstance = new validator(options);
+  validatorInstance.validate(['appveyor.yml']);
+
+  test.equal(validatorInstance.inValidFilesCount, 1);
+});
+
+tape('Test creation of JSON file from YAML structure', test => {
+  test.plan(1);
+
+  const options = {
+    writeJson: true
+  }
+
+  const validatorInstance = new validator(options);
+  validatorInstance.validate(['appveyor.yml']);
+
+  fs.unlink('appveyor.json', function (err) {
```

Comment:
@paazmaya `fs.unlink` returns an error if the file (to be deleted) does not exist. Hence it performs the role of checking and if found, deleting the file.

I have added another commit making changes requested by you. Please have a look at them.